app/assets/javascripts/components/StickyColumn.js

```diff
@@ -0,0 +1,215 @@
+
+  /**
+   * Recalculate offsets based on content height and sticky column's contents
+   */
+  StickyColumn.prototype._measureDom = function() {
+    this.$parent = $(this.config.selectors.parent);
+    this.contentHeight = $(this.config.selectors.mainContent).height();
+    this.topMargin = parseInt(this.$parent.css('marginTop'), 10);
+    this.top = this.$parent.offset().top;
+    this.bottom = this.top + this.contentHeight - this.topMargin - this.$el.height();
+
+    if (this.isInSidebar) {
+      this._showInSidebar();
+    }
+  };
+
+  /**
+   * Works out whether the sticky component is in the right side bar or not
+   */
+  StickyColumn.prototype._showElement = function() {
+    if (!this.isInSidebar && this._isElementVisible(this.$el)) {
+      this._showInSidebar();
+      this.isInSidebar = true;
+      return;
+    }
+
+    if (this.isInSidebar && !this._isElementVisible(this.$el)) {
+      this._hide();
+      this.isInSidebar = false;
+    }
+  };
+
+  /**
+   * calculates the height of the parent container and the width of the sticky element
+   */
+  StickyColumn.prototype._showInSidebar = function() {
+    this.$parent.css('height', this.contentHeight - this.topMargin);
+    this.$el.css('width', this.$parent.width());
+  };
+
+  /**
+   * reset the parent container back to it's original height
+   */
+  StickyColumn.prototype._hide = function() {
+    this.$parent.css('height', 'auto');
+  };
+
+  /**
+   * Work out the position of the container in the context of the scroll of the page
+   */
+  StickyColumn.prototype._positionComponent = function() {
+    var scrollTop;
+
+    if (!this.isInSidebar) {
+      return;
+    }
+
+    scrollTop = this._getWindowScroll();
+
+    if (!this.isFixed && !this.isAtBottom && scrollTop > this.top) {
+      this.$el.addClass(this.config.classes.fixed);
+      this.isFixed = true;
+      this._positionComponent();
+      return;
+    }
+
+    if (this.isFixed) {
+      if (scrollTop < this.top) {
+        this.$el.removeClass(this.config.classes.fixed);
+        this.isFixed = false;
+        return;
+      }
+
+      if (scrollTop > this.bottom) {
+        this.$el
+          .removeClass(this.config.classes.fixed)
+          .addClass(this.config.classes.bottom);
+        this.isFixed = false;
+        this.isAtBottom = true;
+        return;
+      }
+    }
+
+    if (this.isAtBottom && scrollTop < this.bottom) {
+      this.$el
+        .addClass(this.config.classes.fixed)
+        .removeClass(this.config.classes.bottom);
+      this.isAtBottom = false;
+      return;
+    }
+  };
+
+  StickyColumn.prototype._getWindowScroll = function() {
+    return $(window).scrollTop();
+  };
+
+  /**
+   * Limit the amount of times a function is called within a set period
+   * @param  {function}
+   * @param  {int}
+   * @return {function}
+   */
+  StickyColumn.prototype._debounce = function(func, wait) {
+    var timeout;
+
+    return function() {
+      var context = this, args = arguments,
```

Comment:
Linting issues – drop `args` to new line and remove spaces before parentheses `later = function ()`